scripts/resolve_network.py

```diff
@@ -192,22 +226,35 @@ def res_constraints(n, snakemake):
 
     lhs = res - electrolysis
 
-    n.model.add_constraints(lhs >= 0, name="RES_annual_matching")
+    n.model.add_constraints(lhs >= 0, name=f"{name} RES_annual_matching")
```

Comment:
@lisazeyen I think I have somewhat isolated the problem: 

1) When I outcomment this line and the "add_constraints" 7 lines below, the model works. So it is indeed the RES matching constraints that are causing the problem (not sure why I was under the impression it was otherwise before).

2) When I use "breakpoint()" below this line and enter "print(n.model.constraints)", it appears to me that "DE1_RES_annual_Matching" has the dimension "Link", while "GlobalConstraint-country_res_constraints_DE" has no dimension. Maybe this is causing the problem? Logically, this constraint should have no dimension. How can I get rid of it? I couldn't spot the difference between these lines and the GlobalConstraint-country_res_constraints.